File: Python/Kode Program/PlayerTCG.py
from Individu import Individu
from Pokemon import Pokemon

class PlayerTCG(Individu):
    def __init__(self, id="", namaPanggilan="", noKTP="", nama="", alamat=""):
        super().__init__(noKTP, nama, alamat)
        self.id = id
        self.namaPanggilan = namaPanggilan
        self.deckList = []

    def setID(self, i): self.id = i
    def setNamaPanggilan(self, np): self.namaPanggilan = np

    def getID(self): return self.id
    def getNamaPanggilan(self): return self.namaPanggilan
    def getDeckList(self): return self.deckList

    def addPokemon(self, p: Pokemon): self.deckList.append(p)

    def printDeck(self):
        print(f"Deck {self.namaPanggilan}:")
        for p in self.deckList:
            p.printInfo()

    def printInfo(self):
        print(f"Player ID: {self.id}, Nama Panggilan: {self.namaPanggilan}")
        super().printInfo()

File: Java/Kode Program/PlayerTCG.java
import java.util.ArrayList;

public class PlayerTCG extends Individu {
    private String id;
    private String namaPanggilan;
    private ArrayList<Pokemon> deckList = new ArrayList<>();

    public PlayerTCG() {}

    public PlayerTCG(String id, String namaPanggilan, String noKTP, String nama, String alamat) {
        super(noKTP, nama, alamat);
        this.id = id;
        this.namaPanggilan = namaPanggilan;
    }

    public void setID(String i) { this.id = i; }
    public void setNamaPanggilan(String np) { this.namaPanggilan = np; }

    public String getID() { return id; }
    public String getNamaPanggilan() { return namaPanggilan; }
    public ArrayList<Pokemon> getDeckList() { return deckList; }

    public void addPokemon(Pokemon p) { deckList.add(p); }

    public void printDeck() {
        System.out.println("Deck " + namaPanggilan + ":");
        for (Pokemon p : deckList) {
            p.printInfo();
        }
    }

    @Override
    public void printInfo() {
        System.out.println("Player ID: " + id + ", Nama Panggilan: " + namaPanggilan);
        super.printInfo();
    }
}
